source/geometry_model/two_merged_chunks.cc

```diff
@@ -0,0 +1,1090 @@
+      std::vector<unsigned int> upper_rep_vec(upper_repetitions, upper_repetitions+dim);
+
+      // The two triangulations that will be merged into coarse_grid.
+      Triangulation<dim> lower_coarse_grid;
+      Triangulation<dim> upper_coarse_grid;
+
+      // Create the lower box.
+      GridGenerator::subdivided_hyper_rectangle (lower_coarse_grid,
+                                                 lower_rep_vec,
+                                                 point1,
+                                                 point4,
+                                                 false);
+
+      // Create the upper box.
+      GridGenerator::subdivided_hyper_rectangle (upper_coarse_grid,
+                                                 upper_rep_vec,
+                                                 point3,
+                                                 point2,
+                                                 false);
+
+      // Merge the lower and upper mesh into one coarse_grid.
+      // Now we have at least two cells.
+      GridGenerator::merge_triangulations(lower_coarse_grid,
+                                          upper_coarse_grid,
+                                          coarse_grid);
+
+      // Transform box into spherical chunk
+      GridTools::transform (
+        [&](const Point<dim> &p) -> Point<dim>
+      {
+        return manifold.push_forward(p);
+      },
+      coarse_grid);
+
+      // Deal with a curved mesh
+      // Attach the real manifold to slot 15.
+      coarse_grid.set_manifold (15, manifold);
+      for (const auto &cell : coarse_grid.active_cell_iterators())
+        cell->set_all_manifold_ids (15);
+
+      // Set the boundary indicators.
+      set_boundary_indicators(coarse_grid);
+      // Make sure the right boundary indicators are set after refinement
+      // through the function set_boundary_indicators above.
+      coarse_grid.signals.post_refinement.connect (
+        [&]()
+      {
+        this->set_boundary_indicators(coarse_grid);
+      });
+    }
+
+
+
+    template <int dim>
+    void
+    TwoMergedChunks<dim>::
+    set_boundary_indicators (parallel::distributed::Triangulation<dim> &triangulation) const
+    {
+      // Iterate over all active cells and (re)set the boundary indicators.
+      for (typename Triangulation<dim>::active_cell_iterator
+           cell = triangulation.begin_active();
+           cell != triangulation.end();
+           ++cell)
+        {
+          // First set the default boundary indicators.
+          for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+            if (cell->face(f)->at_boundary())
+              cell->face(f)->set_boundary_id (f);
+
+          if (cell->face(3)->at_boundary())
+            // Set the upper part of the eastern boundary to indicator 2*dim+1.
+            if ((cell->vertex(GeometryInfo<dim-1>::vertices_per_cell-1).norm() + cell->vertex(0).norm()) / 2.0 > point3[0])
```

Comment:
this probably works, but I find it very awkward to read. I think you are looking for the following: `cell->face(3)->center(true)[0] > point3[0]`?

Same below.